### PR TITLE
Install libnginx-mod-http-passenger on newer systems

### LINKED ISSUE #1340: libnginx-mod-http-passenger should be installed instead of passenger
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: Any
- Ruby: Any
- Distribution: Debian 9 and Ubuntu 18.04+
- Module version: Any

## How to reproduce (e.g Puppet code you use)

add the following to hiera on a debian 9 or Ubuntu 18.04 and apply then nginx module
```
nginx::package_source: passenger
nginx::package_name: nginx-extras
nginx::http_cfg_append:
  'passenger_root': '/usr/lib/ruby/vendor_ruby/phusion_passenger/locations.ini'
  'passenger_ruby': '/usr/bin/passenger_free_ruby
```
## What are you seeing

Nginx throws an error regarding passenger_root
```
nginx: configuration file /etc/nginx/nginx.conf test failed
nginx: [emerg] unknown directive "passenger_root" in /etc/nginx/nginx.conf:64
```

## What behaviour did you expect instead

A valid nginx configuration thats able to understand passenger.

## Any additional information you'd like to impart

The passenger install guide says to use libnginx-mod-http-passenger and not passenger for Debian 9 and Ubuntu 18.04. libnginx-mod-http-passenger will pull passenger as a dependency.

https://www.phusionpassenger.com/docs/tutorials/deploy_to_production/installations/oss/ownserver/ruby/nginx/

### PR DESCRIPTION
**Pull request by Jonasee**
#### Pull Request (PR) description
On Debian 9 & Ubuntu 18.04 libnginx-mod-http-passenger needs to be
installed instead of passenger according to the installation guide by
passenger. libnginx-mod-http-passenger has a dependency on passenger, so
it will be installed. Without libnginx-mod-http-passenger nginx will not
be able to handle passenger.

Installation guide:
https://www.phusionpassenger.com/docs/tutorials/deploy_to_production/installations/oss/ownserver/ruby/nginx/

#### This Pull Request (PR) fixes the following issues
Fixes #1340

### CODE REVIEW

File: manifests/package/debian.pp [hunk 1]
```diff
@@ -71,9 +71,19 @@
...
         }
 
-        package { 'passenger':
-          ensure  => $passenger_package_ensure,
-          require => Exec['apt_update'],
+        # From Ubuntu 18.04/Debian 9 libnginx-mod-http-passenger is needed
+        # libnginx-mod-http-passenger will install passenger as a dependency
+        if ( ($::operatingsystem == 'Ubuntu') and ($facts['os']['release']['major'] >='18')) or
+           ( ($::operatingsystem == 'Debian') and ($facts['os']['release']['major'] >='9')) {
+              package { 'libnginx-mod-http-passenger':
```
Comment (ekohl) [inline manifests/package/debian.pp]: Since this block is very similar to the previous block (besides the package name), could you use a variable?

**Comment by Jonasee** [comment]
So something like what I just added with the passenger package name in params.pp?

File: manifests/package/debian.pp (hunk 1 above)
Comment (Jonasee) [reply manifests/package/debian.pp]: So something like what I just added with the passenger package name in params.pp?

File: manifests/package/redhat.pp [hunk 2]
```diff
@@ -45,7 +46,7 @@
         }
 
         if $purge_passenger_repo {
-          yumrepo { 'passenger':
+          yumrepo { $passenger_package_name:
```
Comment (ekohl) [inline manifests/package/redhat.pp]: This is a yumrepo, not a package

File: manifests/package/redhat.pp [hunk 3]
```diff
@@ -63,7 +64,7 @@
...
```
Comment (ekohl) [inline manifests/package/redhat.pp]: Same here

File: manifests/package/redhat.pp (hunk 2 above)
Comment (Jonasee) [reply manifests/package/redhat.pp]: fixed

File: manifests/package/redhat.pp (hunk 3 above)
Comment (Jonasee) [reply manifests/package/redhat.pp]: fixed

File: manifests/init.pp [hunk 4]
```diff
@@ -165,6 +165,7 @@
   Boolean $mime_types_preserve_defaults                      = false,
   Optional[String] $repo_release                             = undef,
   $passenger_package_ensure                                  = 'present',
+  $passenger_package_name                                    = $nginx::params::passenger_package_name,
```
Comment (ekohl) [inline manifests/init.pp]:
```suggestion
  String $passenger_package_name                              = $nginx::params::passenger_package_name,
```

**Comment by Jonasee** [comment]
I looked in to that error and it seems that nginx-extras package on Ubuntu 18.04 and Debain 9 adds the /etc/nginx/modules-enabled/ and /etc/nginx/modules-available folders to load dynamic modules. The configuration files are include by default in the nginx configuration but not in the one generated by puppet.

It could be solved by adding loading the passenger module using the dynamic_modules parameter for those versions in the tests or by including /etc/nginx/modules-enabled/*.conf by default. But neither options seems like the correct solution.

**Pull request closed without merging**